src/CountBase.ts

```
@@ -26,72 +34,139 @@ export function createCountBase(disklet: Disklet, databaseName: string, bucketSi
-
             })
+            .then(() => null)
+            .catch(error => {
+              return new Error(`Could not insert data. ${error}`)
+            })
+        } else {
+          return disklet
+            .setText(
+              `${databaseName}/${partition}/${bucketNumber}.json`,
+              JSON.stringify([data])
+            )
+            .then(() => {
+              configData.partitions[partition] = partitionMetadata
+              return disklet.setText(
+                `${databaseName}/config.json`,
+                JSON.stringify(configData)
+              )
+            })
+            .then(() => null)
+            .catch(error => {
+              return new Error(`Could not insert data. ${error}`)
+            })
+        }
+      },
+      query(
+        partition: string = '/',
+        rangeStart: number = 0,
+        rangeEnd: number = rangeStart
+      ): Promise<any[]> {
+        // check that partition only contains letters, numbers, and underscores
+        // sanity check the range
+        const partitionMetadata = configData.partitions[partition]
+        if (partitionMetadata === undefined) {
+          return Promise.reject(
+            new Error(`Partition ${partition} does not exist.`)
+          )
+        }
+        if (partitionMetadata.length === 0) {
+          return Promise.reject(new Error(`Partition ${partition} is empty.`))
+        }
+        const bucketFetchers = []
+        for (
+          let bucketNumber = Math.floor(rangeStart / bucketSize);
+          bucketNumber <= Math.floor(rangeEnd / bucketSize);
+          bucketNumber++
+        ) {
+          bucketFetchers.push(
+            disklet
+              .getText(`${databaseName}/${partition}/${bucketNumber}.json`)
+              .then(rawBucketData => JSON.parse(rawBucketData))
+              .catch(error => {
+                console.log(
+                  `Error getting data from bucket ${bucketNumber}. ${error}`
```

Need to re-throw the error otherwise it gets ignored